### PR TITLE
[Unity] fp16 A x int B GEMM update - support int8, more bias shape

### PR DESCRIPTION
**Pull request by masahi**
Update the FT kernel integration to incorporate the following changes:
https://github.com/tlc-pack/cutlass_fpA_intB_gemm/pull/3
https://github.com/tlc-pack/cutlass_fpA_intB_gemm/pull/4

It also contains other misc changes to better support dolly and gptj (dynamic-size layer norm, residual check condition for dynamic-shape residual etc).

@vinx13 @sunggg @yzh119

### CODE REVIEW

File: tests/python/relax/test_codegen_cutlass.py [hunk 1]
```diff
@@ -1488,6 +1491,153 @@ def split_transform_deploy_mod(mod):
...
+                )
+                lv3: R.Tensor((64,), dtype="float16") = lv[1]
+                lv4: R.Tensor((64, 64), dtype="int8") = R.builtin.stop_lift_params(lv2)
+                lv5: R.Tensor((64,), dtype="float16") = R.builtin.stop_lift_params(lv3)
+                lv6 = R.call_tir(
+                    cls.decode, (lv4, lv5), out_sinfo=R.Tensor((64, 64), dtype="float16")
+                )
+                lv1_1: R.Tensor((64, 64), dtype="float16") = R.matmul(x, lv6, out_dtype="float16")
+                lv2_1: R.Tensor((64, 128), dtype="float16") = R.add(lv1_1, bias)
+                lv2_2: R.Tensor((64, 128), dtype="float16") = R.nn.gelu(lv2_1)
```
Comment (masahi) [inline tests/python/relax/test_codegen_cutlass.py:1590]: This test demonstrates that we are no supporting a bias shape like this and also gelu activation offloaded to FT.

File: python/tvm/contrib/cutlass/gen_tensor_op.py [hunk 2]
```diff
@@ -792,6 +801,12 @@ def get_batch_on_arg(arg_name, arg_shape):
         headers.append("cutlass/layout/matrix.h")
         attrs = {"input": func_args[0], "gamma": func_args[1], "beta": func_args[2]}
         attrs.update(dict(annotations))
+
+        if isinstance(attrs["M"], tvm.tir.Var):
```
Comment (sunggg) [inline python/tvm/contrib/cutlass/gen_tensor_op.py:805]: Does this mean we support batch?

**Review by sunggg — APPROVED** [review]
LGTM. One clarification question.

File: python/tvm/contrib/cutlass/gen_tensor_op.py (hunk 2 above)
Comment (masahi) [reply python/tvm/contrib/cutlass/gen_tensor_op.py:805]: This is unrelated to this PR but needed for dolly. It enables layer norm offload with dynamic first dimension.

**Review by vinx13 — APPROVED** [review]

**Pull request merged**